adaptor/elasticsearch/clients/v5/writer.go

```diff
@@ -90,10 +91,47 @@ func (w *Writer) Write(msg message.Msg) func(client.Session) (message.Msg, error
 			w.bp.Flush()
 			br = elastic.NewBulkDeleteRequest().Index(w.index).Type(indexType).Id(id)
 		case ops.Insert:
-			br = elastic.NewBulkIndexRequest().Index(w.index).Type(indexType).Id(id).Doc(msg.Data())
+			if _, ok := msg.Data()["elastic_parent"]; ok {
+
+				var parent_id string
+				switch msg.Data()["elastic_parent"].(type) {
+				case string:
+					parent_id = msg.Data()["elastic_parent"].(string)
+				case bson.ObjectId:
+					var parent bson.ObjectId
+					parent = msg.Data()["elastic_parent"].(bson.ObjectId)
+					parent_id = parent.Hex()
+				default:
+					parent_id = fmt.Sprintf("%v", msg.Data()["elastic_parent"])
+				}
+
+				br = elastic.NewBulkIndexRequest().Index(w.index).Type(indexType).Id(id).Parent(parent_id).Doc(msg.Data())
+				msg.Data().Delete("elastic_parent")
+			}else{
+
+				br = elastic.NewBulkIndexRequest().Index(w.index).Type(indexType).Id(id).Doc(msg.Data())
```

Comment:
if we can pull out the `switch` block, then this logic could be simplified as such:

```go
indexReq := elastic.NewBulkIndexRequest().Index(w.index).Type(indexType).Id(id)
if parent_id, ok := msg.Data()["elastic_parent"]; ok {
    indexReq.Parent(parend_id)
    msg.Data().Delete("elastic_parent")
}
indexReq.Doc(msg.Data())
br = indexReq
```

and similar could be achieved in the `ops.Update` block.